R/num.R

```diff
@@ -185,7 +185,7 @@ vec_arith.pillar_num <- function(op, x, y, ...) {
 vec_arith.pillar_num.default <- function(op, x, y, ...) {
   "!!!!DEBUG vec_arith.pillar_num.default(`v(op)`)"
   stopifnot(is.numeric(x), is.numeric(y))
-  out <- vec_arith_base(op, x, y)
+  out <- vec_arith_base(op, vec_proxy(x), vec_proxy(y))
```

Comment:
Not sure this is quite right, but I don't have any better advice for generic `vec_arith()` at the moment. Since you check for `is.numeric()` above, it feels a bit unnecessary to take the proxy because you're restricting the inputs to have numeric storage.

Also conceptually there is no general guarantee that `vec_proxy()` is the adequate numeric storage for arithmetic operations. For instance the proxy might be a data frame (record type) and the numeric storage might be in a column of that data frame. For the moment though, we don't have anything better in vctrs.